### DISCUSSION TITLE
Visualization Ai2THOR scenes on Ubuntu

### DISCUSSION THREAD

**Discussion by moorthyrahul** [allenai/ai2thor#890] [Q&A]
Hey, I'm trying Ai2THOR for the first time. I installed it via pip on Ubuntu 18.04. After installing, I wanted to verify the installation using the following (via python3.7 terminal):

from ai2thor.controller import Controller
controller = Controller(scene="FloorPlan10")
event = controller.step(action="RotateRight")

There's no output after the controller command and I'm unable to visualize the scene. Also in the locally downloaded ~/.ai2thor directory, I don't see any game environment containing the 3D scenes. Has anyone else faced this issue?

Thanks in advance!

**Comment by mattdeitke** [comment]
Hi @moorthyrahul,

Can you try something like printing the event:
```python
from ai2thor.controller import Controller
controller = Controller(scene="FloorPlan10")
event = controller.step(action="RotateRight")
print(event)
```
or saving the RGB image frame:
```python
from ai2thor.controller import Controller
from PIL import Image
controller = Controller(scene="FloorPlan10")
event = controller.step(action="RotateRight")
Image.fromarray(event.frame).save("frame.png")

**Reply by moorthyrahul** [reply]
Hi @mattdeitke,

Thank you for your response. I tried that, but still unable to view the output scene from the controller (below image) and there's no further feedback. I'm using python 3.6, Ubuntu 18.04 and installed Ai2THOR via pip install.

![Screenshot from 2021-09-22 13-19-28](https://user-images.githubusercontent.com/84152860/134416102-6e661fec-d756-4fc4-bd3b-788bfbf1b133.png)

Kindly let me know if there's a setup issue from my end!

**Reply by mattdeitke** [reply]
Are you running on a headless machine (i.e., a machine without a monitor attached) by chance?

**Reply by moorthyrahul** [reply]
I'm running Ubuntu via a virtual machine

**Reply by 1625368821** [reply]
I have the same trouble.How to solve this issue for you?

**Comment by victorming** [comment]
Same issue, anybody can help?
